targets/TARGET_STM/gpio_irq_api.c

```diff
@@ -194,6 +195,8 @@ int gpio_irq_init(gpio_irq_t *obj, PinName pin, gpio_irq_handler handler, uint32
         return -1;
     }
 
+    core_util_critical_section_enter();
```

Comment:
Please also protect also the free function ...
Also please add a comment why only init (& free) need to be protected (because other functions are protected at upper level (mbed driver or hal level)

Reply:
critical section and clear flag added in the free function

Reply:
Getting back to Pull Request message s not always handy ... I'd like to see the below information either in the code comments, or in commit message (or both)

> We are adding a critical section in gpio_irq init and free functions to protect shared code structures
> Note that other functions are protected in API level in InterruptIn

Reply:
Commit message updated!
Thx